File: src/dataset/javascript/triangle_exercism-ThomasZumsteg/gemini/GeminiFlash_triangle_v3.js
/* Classifies triangles */
var Triangle = function(a,b,c) {
    this.triangle = [a,b,c].sort((x, y) => x - y);
};

Triangle.prototype.kind = function() {
    var a = this.triangle[0];
    var b = this.triangle[1];
    var c = this.triangle[2];

    if (a <= 0) {
        throw "Only positive length sides are allowed";
    }

    if (a + b <= c) {
        throw "Not a valid triangle";
    }

    if (a === b) {
        if (b === c) {
            return "equilateral";
        }
        return "isosceles";
    } else if (b === c) {
        return "isosceles";
    }

    return "scalene";
};

module.exports = Triangle;

File: src/dataset/Java/triangle_exercism-ThomasZumsteg/gemini/GeminiFlash_Triangle_v3.java
import java.util.Arrays;

public class Triangle {
    private final double a, b, c;

    public Triangle(double a, double b, double c) throws TriangleException {
        if (a <= 0 || b <= 0 || c <= 0) {
            throw new TriangleException();
        }

        if (a + b <= c || a + c <= b || b + c <= a) {
            throw new TriangleException();
        }

        double[] sides = {a, b, c};
        Arrays.sort(sides);
        this.a = sides[0];
        this.b = sides[1];
        this.c = sides[2];
    }

    public TriangleKind getKind() {
        if (a == c) {
            return TriangleKind.EQUILATERAL;
        } else if (a == b || b == c) {
            return TriangleKind.ISOSCELES;
        } else {
            return TriangleKind.SCALENE;
        }
    }
}